paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerAdapterDataSource.kt

```diff
@@ -51,4 +98,44 @@ internal class CustomerAdapterDataSource @Inject constructor(
     override suspend fun retrieveSetupIntentClientSecret(): CustomerSheetDataResult<String> {
         return customerAdapter.setupIntentClientSecretForCustomerAttach().toCustomerSheetDataResult()
     }
+
+    private suspend fun fetchElementsSession(): Result<ElementsSession> {
+        val paymentMethodTypes = createPaymentMethodTypes()
+        val initializationMode = PaymentSheet.InitializationMode.DeferredIntent(
+            PaymentSheet.IntentConfiguration(
+                mode = PaymentSheet.IntentConfiguration.Mode.Setup(),
+                paymentMethodTypes = paymentMethodTypes,
+            )
+        )
+
+        return elementsSessionRepository.get(
+            initializationMode,
+            customer = null,
+            externalPaymentMethods = emptyList(),
+            defaultPaymentMethodId = null,
+        ).onFailure {
+            errorReporter.report(
+                errorEvent = ErrorReporter.ExpectedErrorEvent.CUSTOMER_SHEET_ELEMENTS_SESSION_LOAD_FAILURE,
+                stripeException = StripeException.create(it)
+            )
+        }
+    }
+
+    private suspend fun fetchInitialPaymentMethods(): Result<List<PaymentMethod>> {
+        return retrievePaymentMethods().toResult().onFailure {
+            errorReporter.report(
+                errorEvent = ErrorReporter.ExpectedErrorEvent.CUSTOMER_SHEET_PAYMENT_METHODS_LOAD_FAILURE,
```

Comment:
is there a corresponding success event for this being logged somewhere?

Reply:
Don't think we created one for this event. Can add it in this PR.